### PR TITLE
DEVDOCS-4651: [new] Buy Online Pick Up in Store Implementation

### PR DESCRIPTION
**Pull request by bc-andreadao**
# [DEVDOCS-4651]

Related to [PR 1729](https://github.com/bigcommerce/dev-docs/pull/1729) (api docss) and [PR 172](https://github.com/bigcommerce-labs/next-dev-center-prototype/pull/172) (redirects)

[DEVDOCS-4651]: https://bigcommercecloud.atlassian.net/browse/DEVDOCS-4651?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by bc-andreadao** [comment]
A note about the linting errors- The times that get returned from the API are `08:00`, but fails the check. I didn't find a name of a time format that I could use in the [OAS3.0 spec file](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.0.0.md#data-types). Could use some help, or I'll just move the format to the description to pass the check.

File: reference/pickup.v3.yml [hunk 1]
```diff
@@ -0,0 +1,393 @@
...
+      operationId: get-pickup
+      tags:
+        - Pickup
+      summary: Get Pickups
+      description: |-
+        Retrieves a list of pickups.
+
+        ## Examples
+
+        This list can be filtered to return pickup objects specific to a list of requested pickup or order IDs.
```
Comment (bc-traciporter) [inline reference/pickup.v3.yml:25]:
```suggestion
        You can filter the list to return pickup objects specific to a list of requested pickup or order IDs. 
```

File: reference/pickup.v3.yml [hunk 2]
```diff
@@ -0,0 +1,393 @@
...
+            application/problem+json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+    delete:
+      operationId: delete-pickup
+      tags:
+        - Pickup
+      summary: Delete Pickups
+      description: |-
+        Delete pickups via providing a list of `pickup_id`s.
```
Comment (bc-traciporter) [inline reference/pickup.v3.yml:121]:
```suggestion
        Delete pickups by providing a list of `pickup_id`s.
```

File: reference/pickup.v3.yml [hunk 3]
```diff
@@ -0,0 +1,393 @@
...
+            type: string
+      responses:
+        '204':
+          description: No Content
+    put:
+      operationId: put-pickup
+      tags:
+        - Pickup
+      summary: Update Pickups
+      description: 'Update pickups by providing `pickup_id`, `collected_at` and `ready_at` in the input.'
```
Comment (bc-traciporter) [inline reference/pickup.v3.yml:150]:
```suggestion
      description: 'Update pickups by providing `pickup_id`, `collected_at`, and `ready_at` in the input.'
```

File: reference/pickup.v3.yml [hunk 4]
```diff
@@ -0,0 +1,393 @@
...
+                properties:
+                  data:
+                    type: array
+                    items:
+                      $ref: '#/components/schemas/Pickup'
+                  meta:
+                    type: object
+        '422':
+          description: |
+            Incorrect entity. Locations was not valid. This is the result of missing required fields, or of invalid data or partial error. See the response for more details.
```
Comment (bc-traciporter) [inline reference/pickup.v3.yml:181]:
```suggestion
            Incorrect entity. Locations were not valid. This results from missing required fields, invalid data, or partial error. See the response for more details.
```

File: reference/pickup.v3.yml [hunk 5]
```diff
@@ -0,0 +1,393 @@
...
+              items:
+                $ref: '#/components/schemas/PickupItem'
+    BasePickupItem:
+      type: object
+      properties:
+        order_product_id:
+          type: integer
+          minimum: 1
+          example: 4
+          description: ID of product.
```
Comment (bc-traciporter) [inline reference/pickup.v3.yml:238]:
```suggestion
          description: ID of product.
```
```suggestion
          description: The ID of the product.
```

File: reference/pickup.v3.yml [hunk 6]
```diff
@@ -0,0 +1,393 @@
...
+          description: Quantity of an item.
+    PickupItem:
+      allOf:
+        - type: object
+          properties:
+            id:
+              type: integer
+              minimum: 1
+              example: 3
+              description: ID of pickup.
```
Comment (bc-traciporter) [inline reference/pickup.v3.yml:252]:
```suggestion
              description: The ID of pickup.
```

File: reference/pickup.v3.yml [hunk 7]
```diff
@@ -0,0 +1,393 @@
...
+        - $ref: '#/components/schemas/BasePickupItem'
+        - type: object
+          properties:
+            status:
+              type: string
+              enum:
+                - AWAITING_COLLECTION
+                - COLLECTED
+              example: COLLECTED
+              description: Status of pickup.
```
Comment (bc-traciporter) [inline reference/pickup.v3.yml:262]:
```suggestion
              description: The status of the pickup.
```

File: reference/pickup.v3.yml [hunk 8]
```diff
@@ -0,0 +1,393 @@
...
+            collected_at:
+              type: string
+              format: date-time
+              example: '2022-08-12T04:15:19.150539Z'
+              description: Datetime when pickup was collected.
```
Comment (bc-traciporter) [inline reference/pickup.v3.yml:267]:
```suggestion
              description: Date/time when the pickup was collected.
```

File: reference/pickup.v3.yml [hunk 9]
```diff
@@ -0,0 +1,393 @@
...
+                Unix: Represents the time in seconds that has elapsed since Unix epoch (00:00:00 UTC on 1 January 1970)
+                ISO-8601: Represents the time in ISO format.
+                Examples include `1660555844` and '2022-06-11T00:00:20Z'.
+            collected_at:
+              type: string
+              description: |
+                Can be provided as two inputs.
+                Unix: Represents the time in seconds that has elapsed since Unix epoch (00:00:00 UTC on 1 January 1970).
+                ISO-8601: Represents the time in ISO format.
+                If this value is not supplied the pickup item collected_at property will be null.
```
Comment (bc-traciporter) [inline reference/pickup.v3.yml:287]:
```suggestion
                If this value is not supplied, the pickup item collected_at property will be null.
```

File: reference/pickup.v3.yml [hunk 10]
```diff
@@ -0,0 +1,393 @@
...
+            - ready_at
+            - pickup_items
+    PostRequestPickupItem:
+      type: object
+      properties:
+        order_product_id:
+          type: integer
+          minimum: 1
+          example: 4
+          description: ID of product.
```
Comment (bc-traciporter) [inline reference/pickup.v3.yml:305]:
```suggestion
          description: The ID of the product.
```

File: reference/pickup.v3.yml [hunk 11]
```diff
@@ -0,0 +1,393 @@
...
+        quantity:
+          type: integer
+          minimum: 1
+          example: 5
+          description: Quantity of an item.
```
Comment (bc-traciporter) [inline reference/pickup.v3.yml:310]:
```suggestion
          description: The quantity of an item.
```

File: reference/pickup.v3.yml [hunk 12]
```diff
@@ -0,0 +1,393 @@
...
+            Unix: Represents the time in seconds that has elapsed since Unix epoch (00:00:00 UTC on 1 January 1970)
+            ISO-8601: Represents the time in ISO format.
+            Examples include `1660555844` and '2022-06-11T00:00:20Z'.
+        collected_at:
+          type: string
+          description: |
+            Can be provided as two inputs.
+            Unix: Represents the time in seconds that has elapsed since Unix epoch (00:00:00 UTC on 1 January 1970).
+            ISO-8601: Represents the time in ISO format.
+            If this value is not supplied the pickup item collected_at property will be null.
```
Comment (bc-traciporter) [inline reference/pickup.v3.yml:340]:
```suggestion
            If this value is not supplied, the pickup item collected_at property will be null.
```

File: reference/pickup_methods.v3.yml [hunk 13]
```diff
@@ -0,0 +1,286 @@
...
+      operationId: get-pickup-methods
+      tags:
+        - Pickup Methods
+      summary: Get Pickup Methods
+      description: |-
+        Retrieves a list of pickup methods.
+
+        ## Examples
+
+        This list can be filtered to return pickup method objects specific to a list of requested pickup method IDs. This is achieved by appending the query string `?id:in=1,3,5` to the resource `/pickup/methods`.
```
Comment (bc-traciporter) [inline reference/pickup_methods.v3.yml:27]:
```suggestion
        You can filter this list to return pickup method objects specific to a list of requested pickup method IDs. This is achieved by appending the query string `?id:in=1,3,5` to the resource `/pickup/methods`.
```

File: reference/pickup_methods.v3.yml [hunk 14]
```diff
@@ -0,0 +1,286 @@
...
+                    type: object
+    put:
+      operationId: put-pickup-methods
+      tags:
+        - Pickup Methods
+      summary: Update Pickup Methods
+      description: |-
+        Update existing pickup methods.
+
+        This is a batch operation that can update 10 pickup methods objects in one request.
```
Comment (bc-traciporter) [inline reference/pickup_methods.v3.yml:121]:
```suggestion
        This batch operation can update 10 pickup method objects in one request.
```

File: reference/pickup_methods.v3.yml [hunk 15]
```diff
@@ -0,0 +1,286 @@
...
+      summary: Delete Pickup Methods
+components:
+  schemas:
+    BasePickupMethod:
+      type: object
+      properties:
+        location_id:
+          type: integer
+          example: 2
+          description: ID of location.
```
Comment (bc-traciporter) [inline reference/pickup_methods.v3.yml:185]:
```suggestion
          description: The ID of the location.
```

File: reference/pickup_methods.v3.yml [hunk 16]
```diff
@@ -0,0 +1,286 @@
...
+          minLength: 0
+          maxLength: 250
+          example: Bring your ID!
+          description: Instructions for picking up items.
+        collection_time_description:
+          type: string
+          minLength: 0
+          maxLength: 250
+          example: Collect during our opening hours of 0900 - 1700
+          description: Description of collection time.
```
Comment (bc-traciporter) [inline reference/pickup_methods.v3.yml:203]:
```suggestion
          description: The description of collection time.
```

File: reference/pickup_methods.v3.yml [hunk 17]
```diff
@@ -0,0 +1,286 @@
...
+          example: 2
+          deprecated: true
+    PickupMethod:
+      allOf:
+        - type: object
+          properties:
+            id:
+              type: integer
+              example: 1
+              description: ID of pickup method.
```
Comment (bc-traciporter) [inline reference/pickup_methods.v3.yml:215]:
```suggestion
              description: The ID of the pickup method.
```

File: reference/pickup_methods.v3.yml [hunk 18]
```diff
@@ -0,0 +1,286 @@
...
+              type: integer
+              description: The total number of pages in the collection.
+              example: 2
+            links:
+              type: object
+              description: Pagination links for the previous and next parts of the whole collection.
+              properties:
+                previous:
+                  type: string
+                  description: Link to the previous page returned in the response.
```
Comment (bc-traciporter) [inline reference/pickup_methods.v3.yml:249]:
```suggestion
                  description: A link to the previous page is returned in the response.
```

File: reference/pickup_methods.v3.yml [hunk 19]
```diff
@@ -0,0 +1,286 @@
...
+                current:
+                  type: string
+                  description: Link to the current page returned in the response.
```
Comment (bc-traciporter) [inline reference/pickup_methods.v3.yml:252]:
```suggestion
                  description: A link to the current page is returned in the response.
```

File: reference/pickup_methods.v3.yml [hunk 20]
```diff
@@ -0,0 +1,286 @@
...
+                next:
+                  type: string
+                  description: Link to the next page returned in the response.
```
Comment (bc-traciporter) [inline reference/pickup_methods.v3.yml:255]:
```suggestion
                  description: A link to the next page is returned in the response.
```

**Review by bc-traciporter — COMMENTED** [review]
Here are my suggestions.

**Review by bc-traciporter — COMMENTED** [review]
Here are my suggestions.

**Review by valentindellangela — DISMISSED** [review]
👍 The main API changes LGTM

**Review by agchan12 — DISMISSED** [review]
The pickup changes are all fine from an API perspective.

**Review by EgorAleshin — APPROVED** [review]

File: reference/checkouts.sf.yml [hunk 21]
```diff
@@ -1195,18 +1195,29 @@ paths:
...
 
-        **Required Query**
-        * `consignments.availableShippingOptions`
+        ### For **shipping** consignments:
+
+          1. Add a new Consignment to Checkout.
+              * Send a `POST` request to `/consignments` with each shipping address, line items IDs, and quantities. Each address can have its own line item IDs.
+              * Provide a full valid customer address before the order is placed. If not provided, the order placement will fail. 
+              * As part of the request URL make sure to add `include=consignments.availableShippingOptions` to return the available shipping options based on line items and shipping locations. This will return `availableShippingOptions` in the response.
+              * As part of the request URL make sure to add `include=consignments.availableShippingOptions` to return the available shipping options based on the items, the address, and the shipping location. This will return `availableShippingOptions` in the response.
```
[the remark this answers is not in the document]
Comment (bc-andreadao) [reply reference/checkouts.sf.yml:1205]: ✔️ removed duplicate line

File: reference/locations.v3.yml [hunk 22]
```diff
@@ -0,0 +1,1045 @@
...
+    name: BigCommerce
+    url: 'https://www.bigcommerce.com'
+    email: support@bigcommerce.com
+  description: |-
+    On this page:
+    - [Locations](#locations)
+    - [Metafields](#metafields)
+
+    ## Locations
+    By default all BigCommerce stores have at least one location (who's `id` = `1`). Additional locations can be created to to track inventory against using the `/locations` endpoints. Initially we will only support the creation of physical inventory locations, which must have an address.
```
[the remark this answers is not in the document]
Comment (bc-andreadao) [reply reference/locations.v3.yml:18]: updated the suggestion to remove "whose" (convention elsewhere in the doc to not use "whose" for inanimate objects)

**Comment by bc-andreadao** [comment]
This was moved to the SSOT repository, closing PR

**Pull request closed without merging**